js/apps/system/_admin/aardvark/APP/react/src/views/views/arangoSearchView/usePatchArangoSearchView.tsx

```diff
@@ -0,0 +1,117 @@
+import { pick } from "lodash";
+import { getApiRouteForCurrentDB } from "../../../utils/arangoClient";
+import { encodeHelper } from "../../../utils/encodeHelper";
+import { FormState } from "../constants";
+import { useSyncSearchViewUpdates } from "../searchAliasView/useSyncSearchViewUpdates";
+
+export function usePatchArangoSearchView(
+  view: FormState,
+  oldName: string | undefined,
+  setChanged: (changed: boolean) => void
+) {
+  useSyncSearchViewUpdates({ viewName: view.name });
+  const handleSave = async () => {
+    let hasError = false;
+    const isNameChanged = (oldName && view.name !== oldName) || false;
+    try {
+      if (isNameChanged && oldName) {
+        hasError = await putRenameView({ oldName, view });
+      }
+
+      if (!hasError) {
+        await patchViewProperties({ view, oldName, setChanged });
+      }
+    } catch (e: any) {
+      window.arangoHelper.arangoError(
+        "Failure",
+        `Got unexpected server response: ${e.message}`
+      );
+    }
+  };
+  return handleSave;
+}
+
+async function putRenameView({
+  oldName,
+  view
+}: {
+  oldName: string;
+  view: FormState;
+}) {
+  const route = getApiRouteForCurrentDB();
+  let error = false;
+  const { encoded: encodedOldViewName } = encodeHelper(oldName);
+  const result = await route.put(`/view/${encodedOldViewName}/rename`, {
+    name: view.name.normalize()
+  });
+
+  if (result.body.error) {
+    window.arangoHelper.arangoError(
+      "Failure",
+      `Got unexpected server response: ${result.body.errorMessage}`
+    );
+    error = true;
```

Comment:
Out of curiosity, would returning the error itself make it more useful than just a `boolean` ? Maybe we could consider returning the error?

Reply:
Something like this (_this is something that I recently started following_)

```
const {response, error} = await someAPICall();

if(error) {
 // TODO
}

```

Reply:
we are using this boolean to stop the next API call (after rename). The error is not required anywhere as we are already showing it in a notification 

Also, this was existing code in search alias view, I just moved the code around; Can implement your suggestion across both views when we consolidate it! Will keep track